core/utilities.lua

```diff
@@ -231,6 +253,18 @@ utilities.subContent = function (content)
   return out
 end
 
+-- Call `action` on each content node, recursively, including `content` itself.
+-- Not called on leaves, i.e. strings.
+utilities.forEachContentNode = function (content, action)
```

Comment:
I think there is a naming problem here, "content" is not itself a "node" in the sense we talk about nodes elsewhere in the code. Also this seems more like a recursion function (depth) than an iteration function as named. The name of the function should reflect that.

Reply:
I haven't soaked up many naming conventions yet, how else would you call it? The `forEach` part could be renamed to `traverse`. (I personally think that even though it is implemented recursively, it essentially still iterates. So `forEach` isn't a bad name, but I have no strong opinion one way or the other.)

Reply:
How about `utilities.mapContentRecursive()`, or better yet `utilities.walkContent()`? Really what you're doing here is recursing through a deep tree, not looping through a flat array (we have that and call it `utilities.map()` already).

Reply:
`walkContent` it is then.